Added Local Timezone support for contest timings

Chakit22 opened the pull request:
1. Added a local time zone support for user's to view the timing of coding contests based on their timeZone.
2. Implemented moment library for precise date manipulation.
3. Included a reference link to timeanddate.com, allowing users to conveniently explore and compare coding contest times across different time zones.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enhanced the display of contest start times with timezone support and user-friendly date formatting.

- **Refactor**
  - Integrated the `moment` library for improved date and time handling in contest cards.

- **Documentation**
  - Updated the contest card component documentation to reflect changes in time display logic.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Chakit22 commented:
@pranshugupta54 Please check this out !

anur4ag commented:
Hey @Chakit22, I've reviewed it and some changes need to be made :

- [ ] The start and end timings in  `IndividualCards` component should also be according to the users timezone

<img width="445" alt="Screenshot 2024-01-11 at 00 17 49" src="https://github.com/digitomize/digitomize/assets/71564387/16944e65-cf20-49ac-929e-ed03ca2c7e1e">

- [ ] The duration (time left) on each card in `Contest` page and also in `IndividualCards` should also be relative to the user's timezone .

<img width="348" alt="Screenshot 2024-01-11 at 00 18 42" src="https://github.com/digitomize/digitomize/assets/71564387/cb7f0feb-5457-4634-a8aa-091accb1169e">

- [ ] Also as a bonus you can indicate the timezone in addition with displaying the time of contest like january 11, 2024 08:00 pm `IST`. This IST or timezone should be according to the user's timezone

Chakit22 commented:
Are we changing the design of the cards here? @anur4ag as there is just the start timing mentioned on the cards now. Do I have to add the end timing too on the cards ? or just keep it stored for the future?

anur4ag commented:
> Are we changing the design of the cards here? @anur4ag as there is just the start timing mentioned on the cards now. Do I have to add the end timing too on the cards ? or just keep it stored for the future?

There are no plan to change the individual card's design in the near future, However you just have to replicate the stuff you did for the contest page and just update the `startTimeUnix` and `endTimeUnix` in `IndividualCard.jsx` with the user's local time. and also set the `starts in time` accordingly

Chakit22 commented:
@anur4ag Updated the PR.

These are the things I added.

1. Updated the Individual Cards time to display relative to user's timezone.
2. The start and end timings in Individual cards is also according to the user's timezone.
3. The duration is relative value so it doesn't need to be changed. It is regardless of the user's time zone.
4. The start time is also regardless of the user's timezone. It is again a relative value.

I have already tested all the above scenarios by setting my local time zone to some other timezone. It works fine.

Regarding the bonus feature, I'll add it as a new feature and work on it as it requires some more things.

anur4ag commented:
Hey @Chakit22, that's everything we needed 😄,
Thanks a lot for contributions.

Chakit22 commented:
@anur4ag Hey thank you so much !
I learnt a lot in this task.
